modules/claims_api/spec/lib/claims_api/v2/disability_compensation_pdf_mapper_spec.rb

```diff
@@ -541,53 +541,51 @@
-        expect(secondary_relevance).to eq('ABCDEFG')
-      end
-
-      it 'maps the secondary disability name to the primary disability correctly' do
-        disability_name = form_attributes['disabilities'][0]['name']
-        secondary_disability_name = form_attributes['disabilities'][0]['secondaryDisabilities'][0]['name']
-        sd_label = "#{secondary_disability_name} secondary to: #{disability_name}"
-
-        mapper.map_claim
-
-        claim_info = pdf_data[:data][:attributes][:claimInformation]
-
-        secondary_disability_label = claim_info[:disabilities][3][:disability]
-
-        expect(secondary_disability_label).to eq(sd_label)
-      end
-    end
-
-    context '526 section 5, claim info: disabilities, & has conditions attribute' do
-      it 'maps the has_condition related to exposure method correctly' do
-        mapper.map_claim
-
-        has_conditions = pdf_data[:data][:attributes][:exposureInformation][:hasConditionsRelatedToToxicExposures]
-
-        expect(has_conditions).to eq('YES')
+        mapped_names = mapped_disabilities.pluck(:disability).sort
+        mapped_relevance = mapped_disabilities.pluck(:serviceRelevance).sort
+        mapped_date = mapped_disabilities.pluck(:approximateDate).sort
+        mapped_exposure = mapped_disabilities.pluck(:exposureOrEventOrInjury).sort
+        mapped_keys = mapped_disabilities[0].keys
+
+        request_secondary_disabilities = primary_disabilities.pluck('secondaryDisabilities').compact
+        request_disabilities = (primary_disabilities + request_secondary_disabilities).flatten
+        secondary = primary_disabilities.select { _1['secondaryDisabilities'].present? }
+        request_primary_names = primary_disabilities.pluck('name')
+        request_secondary_names = secondary.map do |s|
+          s['secondaryDisabilities'].map do |a|
+            "#{a['name']} secondary to: #{s['name']}"
+          end
+        end.flatten
+        request_names = (request_primary_names + request_secondary_names).sort
+        request_relevance = request_disabilities.pluck('serviceRelevance').sort
+        request_date = request_disabilities.pluck('approximateDate')
+        split_dates = request_date.map do |a|
+          a.split('-')
+        end
+        converted_request_dates = split_dates.map { |a| [a[1], a[2], a[0]].compact.join('/') }.sort
+        request_exposure = request_disabilities.pluck('exposureOrEventOrInjury').sort
+
+        expect(mapped_names).to eq(request_names)
+        expect(mapped_relevance).to eq(request_relevance)
+        expect(mapped_date).to eq(converted_request_dates)
+        expect(mapped_exposure).to eq(request_exposure)
+        expect(has_conditions).to eq(has_toxic_exposure)
+        # TODO: eq is risky here as it forces ALL keys - these may be blank
+        # if the value provided is nil
+        # include was not playing nice when they matched perfectly
+        expect(accepted_fields).to eq(mapped_keys)
```

Comment:
reworked this to dynamically build what we expect, so if we make a change to the json file that serves the "request", we get the rest free of charge - this should cut down on the need to make changes in multiple places (JSON form 526 and spec file) in respect to the hardcoded values. 

a few of these tests seemed to be of duplicate type so elected to overload this one a bit.

I don't like the `eq` here on 588 but `include` wouldn't pass on a 1 <> 1 match. Open to suggestions there but for the sake of time just did an `eq` now.